src/components/Swap/Toolbar/index.tsx

```diff
@@ -169,33 +168,68 @@ export default memo(function Toolbar() {
     return rows
   }, [gasUseEstimateUSD, impact?.percent, impact?.warning, slippage, trade])
 
-  if (inputCurrency == null || outputCurrency == null || error === ChainError.MISMATCHED_CHAINS) {
-    return null
-  }
+  const getCaptionRow = useMemo(() => {
+    if (inputCurrency == null || outputCurrency == null || error === ChainError.MISMATCHED_CHAINS) {
+      return null
+    }
+    return (
+      <StyledExpando
+        title={
+          <ToolbarRow
+            flex
+            justify="space-between"
+            data-testid="toolbar"
+            onClick={maybeToggleOpen}
+            isExpandable={isExpandable}
+          >
+            {caption}
+          </ToolbarRow>
+        }
+        styledTitleWrapper={false}
+        showBottomGradient={false}
+        open={open}
+        onExpand={maybeToggleOpen}
+        maxHeight={16}
+      >
+        <Column>
+          <ToolbarTradeSummary rows={tradeSummaryRows} />
+          <ToolbarOrderRouting trade={trade} />
+        </Column>
+      </StyledExpando>
+    )
+  }, [caption, error, inputCurrency, isExpandable, maybeToggleOpen, open, outputCurrency, trade, tradeSummaryRows])
+
+  const getSwapActionButton = useMemo(() => {
```

Comment:
this isnt a getter function since it returns the value, right? can you rename to `swapActionButton`? or if you meant to make this a function you shoudl use `useCallback` instead

Reply:
but also you dont necessarily need to create children components like this - you can instead just separate these out into their own functional components, which accomplishes the same thing